pkg/splunk/enterprise/indexercluster.go

```diff
@@ -273,6 +288,41 @@ func ApplyIdxcSecret(mgr *indexerClusterPodManager, replicas int32, mock bool) e
 		}
 	}
 
+	/*
+		During the recycle of indexer pods due to an idxc secret change, if there is a container
+		restart(for example if the splunkd process dies) before the operator
+		deletes the pod, the container restart fails due to mismatch of idxc password between Cluster
+		master and that particular indexer.
+
+		Changing the idxc passwords on the secrets mounted on the indexer pods to avoid the above.
+	*/
+	if len(mgr.cr.Status.IdxcPasswordChangedSecrets) > 0 {
+		for podSecretName := range mgr.cr.Status.IdxcPasswordChangedSecrets {
+			podSecret, err := splutil.GetSecretByName(mgr.c, mgr.cr, podSecretName)
+			if err != nil {
+				return fmt.Errorf("Could not read secret %s, reason - %v", podSecretName, err)
+			}
+
+			// Retrieve namespaced scoped secret data in splunk readable format
+			splunkReadableData, err := splutil.GetSplunkReadableNamespaceScopedSecretData(mgr.c, mgr.cr.GetNamespace())
+			if err != nil {
+				return err
+			}
+
+			podSecret.Data["idxc_secret"] = splunkReadableData["idxc_secret"]
+			podSecret.Data["default.yml"] = splunkReadableData["default.yml"]
+
+			_, err = splctrl.ApplySecret(mgr.c, podSecret)
+			if err != nil {
+				return err
+			}
+			scopedLog.Info("idxc password changed on the secret mounted on pod", "Secret on Pod:", podSecretName)
+
+			// Delete entry in map once value changed
+			delete(mgr.cr.Status.IdxcPasswordChangedSecrets, podSecretName)
+		}
+	}
+
```

Comment:
I may be wrong in understanding, but wanted to check, If we take this approach, then we are going to overwrite the existing versioned secret than creating a new version, if so, the real purpose of the versioning is compromised?

Reply:
The operator will overwrite the existing versioned secret mounted on the Pod(which will stay mounted on the Pod until the operator deletes the pod). Upon deletion, the newer versioned secret will be mounted validating one of the benefits of using the versioning of secrets.

Reply:
As discussed, since this approach is deviating from the versioning logic, I think, we need to update the corresponding .md file for now or somewhere as a Cavea(?),  before we take a final formal approach to address this issue.